src/Admin/SiteHealth.php

```diff
@@ -335,29 +335,31 @@ static function ( $available_service ) {
 	 * @return array
 	 */
 	public function page_cache() {
-		$has_page_caching = $this->has_page_caching();
+		$page_cache_status = $this->get_page_cache_status();
 
-		$badge_color = 'orange';
-		$status      = 'recommended';
+		$badge_color = 'red';
+		$status      = is_wp_error( $page_cache_status ) ? 'critical' : $page_cache_status;
 		$label       = __( 'Page caching is not detected', 'amp' );
 
 		$description = '<p>' . esc_html__( 'The AMP plugin performs at its best when page caching is enabled. This is because the additional optimizations performed require additional server processing time, and page caching ensures that responses are served quickly.', 'amp' ) . '</p>';
 
 		/* translators: 1 is Cache-Control, 2 is Expires, and 3 is Age */
 		$description .= '<p>' . sprintf( __( 'Page caching is detected by making three requests to the homepage and looking for %1$s, %2$s, or %3$s HTTP response headers.', 'amp' ), '<code>Cache-Control: max-age=…</code>', '<code>Expires</code>', '<code>Age</code>' );
 
-		if ( is_wp_error( $has_page_caching ) ) {
+		if ( is_wp_error( $page_cache_status ) ) {
 			$error_info = sprintf(
 				/* translators: 1 is error message, 2 is error code */
 				__( 'Unable to detect page caching due to possible loopback request problem. Please verify that the loopback request test is passing. Error: %1$s (Code: %2$s)', 'amp' ),
-				$has_page_caching->get_error_message(),
-				$has_page_caching->get_error_code()
+				$page_cache_status->get_error_message(),
+				$page_cache_status->get_error_code()
 			);
 
 			$description = "<p>$error_info</p>" . $description;
-		} elseif ( true === $has_page_caching ) {
+		} elseif ( 'recommended' === $page_cache_status ) {
+			$badge_color = 'orange';
+			$label       = __( 'Page caching is not detected', 'amp' );
+		} elseif ( 'good' === $page_cache_status ) {
 			$badge_color = 'green';
-			$status      = 'good';
 			$label       = __( 'Page caching is detected', 'amp' );
 		}
 
```

Comment:
Here we can amend `$description` with the details about which caching headers were found, whether page caching plugin was found, and what the median response time was.